queue_processors/queue_processor/tests/test_queue_listener.py

```diff
@@ -111,11 +93,22 @@ def test_on_message_sends_acknowledgement(self):
         self.assertIsInstance(self.mocked_handler.data_ready.call_args[0][0], Message)
 
     def test_on_message_handler_catches_exceptions(self):
-        "Test on_message correctly handles an exception being raised"
-
+        """Test on_message correctly handles an exception being raised"""
         def raise_expected_exception(msg):
             raise Exception(msg)
 
         self.mocked_handler.data_ready.side_effect = raise_expected_exception
         self.listener.on_message(self.headers, {"run_number": 1234567})
         self.mocked_logger.error.assert_called_once()
+
+    def test_on_disconnected(self):
+        """Test correct calls on disconnect"""
+        mock_con = MagicMock()
+        mock_sub = MagicMock()
+        self.mocked_client.connect = mock_con
+        self.mocked_client.subscribe = mock_sub
+
+        self.listener.on_disconnected()
+
+        mock_con.assert_called_once()
+        mock_sub.assert_called_once()
```

Comment:
Need to add 1 more test - when `self.mocked_client.connect.side_effect = ConnectionError`.

This might get stuck in an infinite loop though, you might need a more "advanced" side effect, or another way to break it:

```
def connect_side_effect(connect_mock):
	connect_mock.side_effect = None
	raise ConnectionError

self.mocked_client.connect.side_effect = partial(connect_side_effect, self.mocked_client.connect)
```